### DISCUSSION TITLE
function st_x(geography) does not exist

### DISCUSSION THREAD

**Discussion by Ben1980** [supabase/supabase#12961] [Questions]
I have a issue with ST_X (basicaly all others like ST_Transform and so on) of postgis extension in supabase local. If I'm running this

```
create or replace function fetch_all_log_entries_ordered(p_user_id uuid)
returns table(
        .
        .
	"long" text,
	"lat" text,
        .
        .
)
language plpgsql
as $$
begin
    return query select 
		.
                .
		ST_X(dive_meta_data.position),
		ST_Y(dive_meta_data.position),
		.
                .
    from logbook_entries 
    .
    .
    .;
end;
$$;

select * from fetch_all_log_entries_ordered('fd4f1aab-76af-41b3-95e8-22048f81defa');
```
via pgadmin on a local supabase instance I get the following error:
```
ERROR:  function st_x(geography) does not exist
LINE 7:   ST_X(dive_meta_data.position),
          ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
QUERY:  select 
		logbook_entries.id,
        dive_meta_data.dive_site,
        dive_meta_data.date,
        dive_meta_data.favorite,
        dive_meta_data.rating,
		ST_X(dive_meta_data.position),
		ST_Y(dive_meta_data.position),
```
The table looks the following:
```
create extension postgis with schema extensions;

grant usage on schema extensions to anon;
grant usage on schema extensions to authenticated;

create table "public"."dive_meta_data" (
    .
    .
    "position" geography(POINT),
    .
    .
);

alter table "public"."dive_meta_data" enable row level security;
CREATE UNIQUE INDEX dive_meta_data_pkey ON public.dive_meta_data USING btree (id);
alter table "public"."dive_meta_data" add constraint "dive_meta_data_pkey" PRIMARY KEY using index "dive_meta_data_pkey";
alter table "public"."logbook_entries" add column dive_meta_data_id uuid references "public"."dive_meta_data";

create policy "Dive meta data entries available through owner"
on "public"."dive_meta_data"
as permissive
for all
to authenticated
using ((auth.uid() = user_id));

```

I would appreciate some hints how to resolve this issue.

**Comment by steve-chavez** [comment]
Try prepending the schema when using the function, like `extensions.st_x`. You can also set a search path to the function to avoid this `CREATE FUNCTION .... SET search_path = 'extensions'`.

**Reply by Ben1980** [reply]
Thank you, I'm not sure if I understood it right. Do you think about something like this?

extensions, ST_X(dive_meta_data.position),

Because this seems not to work.

**Reply by lucashaley** [reply]
Try `gis.ST_X()`

**Comment by thawankeane** [comment]
Same issue here, @Ben1980 did you figure out how to fix it?

**Reply by Ben1980** [reply]
No, I do the translation now manually as string operation on client side

**Comment by hacxman** [comment]
Solution to your problem is that you have to typecast from geography to geometry, look:
select ST_X(geometry(someposition)) from sometable;

**Comment by markwcollins** [comment]
this worked for me

```SELECT ST_X(geogcolumn::geometry), ST_Y(geogcolumn::geometry) FROM thetable;```

source: https://gis.stackexchange.com/questions/32680/postgis-convert-geographypoint-4326-to-gps

**Reply by ddavisj** [reply]
Thanks a bunch!
